src/main.ts

```diff
@@ -121,6 +162,15 @@ export class Minifier {
     return Minifier.reservedJSKeywords.hasOwnProperty(str);
   }
 
+  renameProperty(name: string): string {
+    if (!this.renameMap.hasOwnProperty(name)) {
+      let newName = this.generateNextPropertyName(this.lastGeneratedPropName);
```

Comment:
would it be less error prone to do the assignment into `this.lastGeneratedPropName` within the method `generateNextPropertyName`? Then the method would be in control of its invariant, and it would be more obvious how it works.